src/MonoTorrent.Client/MonoTorrent.Client.Modes/StartingMode.cs

```diff
@@ -159,10 +159,16 @@ async Task VerifyHashState ()
             if (!Manager.HasMetadata || !Manager.HashChecked)
                 return;
 
-            // FIXME: I should really just ensure that zero length files always exist on disk. If the first file is
-            // a zero length file and someone deletes it after the first piece has been written to disk, it will
-            // never be recreated. If the downloaded data requires this file to exist, we have an issue.
+
             foreach (ITorrentManagerFile file in Manager.Files) {
+                if (file.Length == 0 && !File.Exists (file.FullPath)) {
+                    Directory.CreateDirectory (Path.GetDirectoryName (file.FullPath));
```

Comment:
This is failing if the `TorrentManager` is started using `StartAsync()` instead of explicitly calling `CheckHashAsync()` because `Manager.HashChecked == false` and it short circuits above. The result is the empty file appears to only be created when explicitly checking the hash